core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```
@@ -596,8 +595,11 @@ public void run() {
 
         @Override
         public void messageReceived(UnicastPingRequest request, TransportChannel channel) throws Exception {
-            channel.sendResponse(handlePingRequest(request));
+            if (request.pingResponse.clusterName().equals(clusterName)) {
+                channel.sendResponse(handlePingRequest(request));
+            }
```

should we throw an exception instead? The channel won't be closed otherwise?